api/v1alpha1/api_validation.go

```diff
@@ -0,0 +1,148 @@
+			case TypePublicKeySSH:
+				name := fmt.Sprintf("Secrets.%s.%s",
+					config.Secrets[secretIndex].Name,
+					config.Secrets[secretIndex].Keys[keyIndex].Name,
+				)
+				// must have privateKeyPath
+				if len(key.PrivateKeyPath) == 0 {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].PrivateKeyPath, name, "PrivateKeyPath", "privateKeyPathNotSet", "")
+					return
+				}
+				// privateKeyPath must be valid
+				if !pathExistsInSecretConfigs(key.PrivateKeyPath, config.Secrets) {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].PrivateKeyPath, name, "PrivateKeyPath", "privateKeyPathNotFound", "")
+				}
+			case TypePKCS12:
+				name := fmt.Sprintf("Secrets.%s.%s",
+					config.Secrets[secretIndex].Name,
+					config.Secrets[secretIndex].Keys[keyIndex].Name,
+				)
+				// must have keystoreAliases
+				if len(key.AliasConfigs) == 0 {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].AliasConfigs, name, "AliasConfigs", "keystoreAliasesNotSet", "")
+					return
+				}
+				// must have keyPassPath
+				if len(key.KeyPassPath) == 0 {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].KeyPassPath, name, "KeyPassPath", "keyPassPathNotFound", "")
+					return
+				}
+				// keyPassPath must be valid
+				if !pathExistsInSecretConfigs(key.KeyPassPath, config.Secrets) {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].KeyPassPath, name, "KeyPassPath", "keyPassPathNotValid", "")
+					return
+				}
+				// must have storePassPath
+				if len(key.StorePassPath) == 0 {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].StorePassPath, name, "StorePassPath", "storePassPathNotFound", "")
+					return
+				}
+				// storePassPath must be valid
+				if !pathExistsInSecretConfigs(key.StorePassPath, config.Secrets) {
+					sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].KeyPassPath, name, "StorePassPath", "storePassPathNotValid", "")
+					return
+				}
+				for aliasIndex, alias := range key.AliasConfigs {
+					switch alias.Type {
+					case TypeCA:
+						name := fmt.Sprintf("Secrets.%s.%s.%s",
+							config.Secrets[secretIndex].Name,
+							config.Secrets[secretIndex].Keys[keyIndex].Name,
+							config.Secrets[secretIndex].Keys[keyIndex].AliasConfigs[aliasIndex].Alias,
+						)
+						// must have passwordPath
+						if len(alias.PasswordPath) == 0 {
+							sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].AliasConfigs[aliasIndex].Alias, name, "CA", "passwordPathNotSet", "")
+							return
+						}
+						// passwordPath must be valid
+						if !pathExistsInSecretConfigs(alias.PasswordPath, config.Secrets) {
+							sl.ReportError(config.Secrets[secretIndex].Keys[keyIndex].AliasConfigs[aliasIndex].Alias, name, "CA", "passwordPathNotFound", "")
+							return
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func pathExistsInSecretConfigs(path []string, secrets []*SecretConfig) bool {
+	found := false
+path:
```

Comment:
has `go fmt` been run here?

Reply:
is part of the make

Reply:
This is a go label.

Reply:
That said, ideally your editor should be setup to go fmt and go vet on every file save. It makes development much easier by auto-indenting code when you move it around, and by pointing out mistakes early on.

Reply:
yea the indentation just looked off to me.